src/OpenTelemetry.Exporter.Console/ConsoleExporterOptions.cs

```diff
@@ -26,10 +25,9 @@ public class ConsoleExporterOptions
         /// </summary>
         public ConsoleExporterOutputTargets Targets { get; set; } = ConsoleExporterOutputTargets.Console;
 
-        /// <summary>
-        /// Gets or sets the metric export interval in milliseconds. The default value is <c>Timeout.Infinite</c>.
-        /// </summary>
-        public int MetricExportIntervalMilliseconds { get; set; } = Timeout.Infinite;
+        public MetricReaderExportType MetricReaderExportType { get; set; } = MetricReaderExportType.Simple;
```

Comment:
@reyang @alanwest Can you share your thoughts on making the default as Periodic? 
One thing I have observed is, users are used to seeing traces/logs from `ConsoleExporter`, by just doing `AddConsoleExporter`. However, for Metrics, it won't work. There is nothing in the spec which says `ConsoleExporter` should be paired with a periodic exporting reader, but I think it makes sense to pair it with `PeriodicExportingReader` by default, to have similar experience as traces/logs.

This example: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/AspNetCore/Startup.cs#L137, won't work as is, and most folks will be puzzled as to why Metrics are not showing up....

Reply:
I do agree that it makes sense for the console exporter to default to periodic. I think the general expectation is that info piped to stdout is emitted automatically vs invoking a collect or flush.

Also, my understanding is the console exporter is primarily useful in testing/troubleshooting scenarios. I think this is especially true for metrics, so seeing a quick validation things are alive and working with the single line `AddConsoleExporter()` is a plus.

That said, I think when we take on the work to support additional output formats, I think the usefulness of the console exporter may expand beyond testing scenarios. In those cases I think folks will be fine configuring the exporter how they need it.

Reply:
I think the challenges are:
1. What should be the default exporting period - I think we don't know at this moment.
2. What is the behavior during exit.

Reply:
Could we borrow that we do for OTLP Metric exporter? 
Same default export interval as OTLP.

The behavior during exit - not sure I follow the specifics here. We do the usual shutdown/dispose right?

Reply:
> Could we borrow that we do for OTLP Metric exporter? Same default export interval as OTLP.

You could. The question is _why_ do you think Console and OTLP should have the same default interval?

I think there should be a concrete usage scenario and we need to think about the design. For example, if the expectation is to use it for **interactive** output, the refresh rate should be `<= 0.3s` (based on the common study from interactive UI, and that's the reason why in the Stress test I picked `0.2s` delay for each refresh).

There are many issues with the current Console exporter. For example if there are both trace and metrics exporter, and folks would want to see Exemplar which connects traces and metrics, the output will be mixed up because the lack of synchronization on the outputs.

My gut feeling is that by making it outputs periodically without understanding the usage scenario and without a good design, is probably going to make the current situation worse.

Reply:
I defer to @reyang's judgment on not yet understanding the usage scenario. I've definitely not thought deep about either interactivity nor metric's relationship to tracing.

That said, if we continue to default to non-periodic, perhaps we update the ASP.NET Core example to use periodic to help guide folks?

Reply:
> That said, if we continue to default to non-periodic, perhaps we update the ASP.NET Core example to use periodic to help guide folks?

Either we use periodic console exporter (and gather feedback about the default interval), or use Prometheus exporter. (the current console metrics exporter has a bug that it only outputs to stdout rather than leveraging the abstraction [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/ca228387628b0783ad8146d63680e77d64555697/src/OpenTelemetry.Exporter.Console/ConsoleExporter.cs#L29))